### PR TITLE
Add qt-webengine recipe

### PR DESCRIPTION
**Pull request by andfoy**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by andfoy** [comment]
This PR uses a git repository since the 5.15.2 tarball published on Qt has several performance errors. Also, other Linux distributions such as Arch Linux use the  `87-based` branch of `qt-chromium` directly, as opposed to a tarball. Finally, since Qt stopped producing official tarballs, our best shot at having the latest version is compiling from the `v5.15.4-lts` branch.

https://archlinuxarm.org/packages/aarch64/qt5-webengine/files/PKGBUILD

**Comment by andfoy** [comment]
@isuruf, we won't be able to build this on the CIs due to the time limit restriction, what do you suggest in order to move forward with the review

**Comment by looooo** [comment]
@conda-forge/core what is the status of this PR?

**Comment by andfoy** [comment]
I'm going to assist to the monthly conda-forge meeting in order to move forward this PR

File: recipes/qt-webengine/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    git_rev: 87-based
+    folder: qtwebengine-chromium
+    patches:
+      - patches/0001-aarch64-sys-state-structs.patch       # [aarch64]
+      - patches/0002-macos-fatal-warnings.patch            # [osx]
+      - patches/0003-macos-arm64-chromium.patch            # [arm64]
+      - patches/0004-macos-arm64-float16-conv.patch        # [arm64]
+      - patches/0005-macos-use-CONDA_BUILD_SYSROOT.patch   # [osx]
+
+  - url: https://download.qt.io/development_releases/gnuwin32/gnuwin32.zip  # [win]
```
Comment (isuruf) [inline recipes/qt-webengine/meta.yaml:22]: Please include the license file of this package in `license`.

File: recipes/qt-webengine/build.sh [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    make install
+fi
+
+if [[ $(uname) == "Darwin" ]]; then
+    export AR=$(basename ${AR})
+    export RANLIB=$(basename ${RANLIB})
+    export STRIP=$(basename ${STRIP})
+    export OBJDUMP=$(basename ${OBJDUMP})
+    export CC=$(basename ${CC})
+    export CXX=$(basename ${CXX})
```
Comment (isuruf) [inline recipes/qt-webengine/build.sh:72]:
```suggestion
```

File: recipes/qt-webengine/build.sh [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+    # Let Qt set its own flags and vars
+    for x in OSX_ARCH CFLAGS CXXFLAGS LDFLAGS
+    do
+        unset $x
+    done
+
+    # Some test runs 'clang -v', but I do not want to add it as a requirement just for that.
+    ln -s "${CXX}" ${HOST}-clang || true
```
Comment (isuruf) [inline recipes/qt-webengine/build.sh:81]:
```suggestion
```

File: recipes/qt-webengine/patches/0004-macos-arm64-float16-conv.patch [hunk 4]
```diff
@@ -0,0 +1,27 @@
+diff --git a/chromium/third_party/skia/src/opts/SkRasterPipeline_opts.h b/chromium/third_party/skia/src/opts/SkRasterPipeline_opts.h
```
Comment (isuruf) [inline recipes/qt-webengine/patches/0004-macos-arm64-float16-conv.patch:1]: Is this an upstream patch? If so, please include a link

**Comment by andfoy** [comment]
# Latest logs
| Arch         |   URL    |
|:-------------:|:----------:|
| `linux-64`  | [build-log](https://drive.google.com/file/d/1OMXJs3zJNvjEA1a1i6Ul0UvFLvCGbQJ7/view?usp=sharing) |
| `linux-aarch64` | pending |
| `win-64` | [build-log](https://drive.google.com/file/d/1R-3FiNgUxrlKf8jhCPEzlT-J3xFXWzwe/view?usp=sharing) |
| `osx-64` | pending |
| `osx-arm64` | pending |

**Comment by andfoy** [comment]
@isuruf, I've just published all the build logs for all the platforms in the comment above:
https://github.com/conda-forge/staged-recipes/pull/17233#issuecomment-1016633307

**Comment by isuruf** [comment]
Is that with or without the last commit?

**Comment by izahn** [comment]
The latest commit has `-jCPU_COUNT`, should be `j$CPU_COUNT`

**Comment by andfoy** [comment]
Yes, they are with the latest changes, I'm running the builds against https://github.com/andfoy/qt-recipe/pull/3, which has the same recipe as this PR but rendered for reproducing purposes

**Comment by izahn** [comment]
While I'm here, can I ask what the timeline and process is for moving packages to `qt-main` etc.? Some packages are already moving over, e.g., https://github.com/conda-forge/qtwebkit-feedstock/pull/18 , some are waiting, e.g., https://github.com/conda-forge/poppler-feedstock/pull/123 , and others may not be thinking about it at all yet. Any guidance on what packages depending on `qt` should be doing would be appreciated.

**Comment by andfoy** [comment]
@isuruf, any updates on this one?

**Comment by izahn** [comment]
> @izahn, my guess is that `qt` would become a metapackage that depends on both `qt-main` and `qtwebengine`?

OK, so then there would be a normal migration to update the `qt` global pin? Is someone working on that? I'm happy to help but I haven't yet learned how to do that.

**Comment by andfoy** [comment]
@isuruf Thanks for the help with this one! I'll ping you to upload the packages once the feedstock is rendered

**Pull request merged**